core-bundle/src/Resources/contao/classes/Theme.php

```diff
@@ -663,9 +723,56 @@ protected function extractThemeFiles($arrFiles, $arrDbFields)
 		$this->import(Automator::class, 'Automator');
 		$this->Automator->generateSymlinks();
 
+		if (($exampleWebsite = Input::post('example_website')) && isset($exampleWebsites[$exampleWebsite]))
+		{
+			$importType = Input::post('example_website_import');
+			$this->importExampleWebsite($exampleWebsites[$exampleWebsite], $importType === 'data_no_truncate', $importType !== 'full');
+		}
+
 		$this->redirect(str_replace('&key=importTheme', '', Environment::get('request')));
 	}
 
+	private function importExampleWebsite(string $exampleWebsite, bool $preserveData, bool $insertOnly): void
+	{
+		$connection = System::getContainer()->get('database_connection');
+
+		$userRow = $connection->fetchAssociative('SELECT * FROM tl_user WHERE id = ?', array((int) BackendUser::getInstance()->id));
+
+		if (!$preserveData && $insertOnly)
+		{
+			$tables = $connection->createSchemaManager()->listTableNames();
+
+			foreach ($tables as $table)
+			{
+				if (0 === strncmp($table, 'tl_', 3))
+				{
+					$connection->executeStatement('TRUNCATE TABLE ' . $connection->quoteIdentifier($table));
+				}
+			}
+		}
+
+		$data = file(Path::join(System::getContainer()->getParameter('kernel.project_dir'), $exampleWebsite));
+
+		try
+		{
+			foreach ($insertOnly ? preg_grep('/^INSERT /', $data) : $data as $query)
+			{
+				$connection->executeStatement($query);
+			}
+		}
+		finally
+		{
+			// Restore backend user
+			if ((int) $connection->fetchOne('SELECT COUNT(*) FROM tl_user') < 1)
```

Comment:
shouldn't we check for admins as well? Also, does it make sense to restore users if their groups were replaced (or truncated)?

Reply:
We should restore the current user so that he does not get logged out.

Reply:
IMO this is pretty important for UX. Otherwise you would need to create a user to enter the back end, import the file and then, again, go back and create a user again. I don't think people would like this.

Reply:
yeah maybe SQL theme import should only be available to admins (maybe it already is)? and we should restore that admin user, then we don't need to know/verify about user groups.

Reply:
Users with permission to import themes are “implicit admins” 
<img width="740" alt="Bildschirmfoto 2022-04-05 um 09 26 15" src="https://user-images.githubusercontent.com/367169/161701046-4486923b-b80e-448c-8fab-4e2de78dc495.png">

Reply:
IMHO, the current logic is correct. If the SQL dump includes back end users, you will be logged out after the import and have to log in again with the user defined in the theme. Otherwise, you can keep working with the user you have already created.